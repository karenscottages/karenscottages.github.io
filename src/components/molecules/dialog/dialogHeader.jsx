import React from 'react';

export const DialogHeader = ({close, title, theme}) => <div className={`p-2 is-flex is-justify-content-space-between card ${theme.contentBackground} ${theme.textColour}`}>
    {title && <h2>{title}</h2>}
    <span onClick={close}><i className='far fa-times-circle mr-1'/></span>
</div>;

