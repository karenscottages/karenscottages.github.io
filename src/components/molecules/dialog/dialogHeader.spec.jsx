import React from 'react';
import renderer from 'react-test-renderer';
import { DialogHeader } from './dialogHeader.jsx';

describe("Molecules/dialog/DialogHeader", () => {
    describe("DialogHeader", () => {
        it('Renders with theme and title', () => {
            const tree = renderer.
                create(<DialogHeader theme={{backgroundColour: 'white', textColour: 'black'}} title="Test Overlay"/>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe("DialogHeader", () => {
        it('Renders with theme and no title', () => {
            const tree = renderer.
                create(<DialogHeader theme={{backgroundColour: 'black', textColour: 'white'}}/>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe("DialogHeader", () => {
        it('Renders with a close function', () => {
            const tree = renderer.
                create(<DialogHeader theme={{backgroundColour: 'black', textColour: 'white'}} close={() => {('Hello World');}}/>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

});