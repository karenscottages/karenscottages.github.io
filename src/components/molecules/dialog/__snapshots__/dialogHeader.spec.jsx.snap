// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/dialog/DialogHeader DialogHeader Renders with a close function 1`] = `
<div
  className="p-2 is-flex is-justify-content-space-between card undefined white"
>
  <span
    onClick={[Function]}
  >
    <i
      className="far fa-times-circle mr-1"
    />
  </span>
</div>
`;

exports[`Molecules/dialog/DialogHeader DialogHeader Renders with theme and no title 1`] = `
<div
  className="p-2 is-flex is-justify-content-space-between card undefined white"
>
  <span>
    <i
      className="far fa-times-circle mr-1"
    />
  </span>
</div>
`;

exports[`Molecules/dialog/DialogHeader DialogHeader Renders with theme and title 1`] = `
<div
  className="p-2 is-flex is-justify-content-space-between card undefined black"
>
  <h2>
    Test Overlay
  </h2>
  <span>
    <i
      className="far fa-times-circle mr-1"
    />
  </span>
</div>
`;
